### DISCUSSION TITLE
Question about User Account Rights in Coonfiguration File

### DISCUSSION THREAD

**Discussion by Domkra** [Ylianst/MeshCentral#5953] [Q&A] [answered]
Hi everyone, I have a question.

Is it possible to take away the rights to create new devices and to lock account settings for new accounts in the configuration file?

I always find th option of  "netools" and "nonewgroups".

If it´s not possible, perhaps it could be implemented similar to the example? 

          "newAccountsRights": [
            "nonewgroups",
            "notools",
            "nonewdevices",
            "lockaccountsettings"
          ],

**Comment by Domkra** [comment]
Hi everyone, I have found a workaround to implement the above mentioned. However, I am still new inexperienced in using GitHub, that´s why I´m writing my changes here.

So that the proposed config part works: 

          "newAccountsRights": [
            "nonewgroups",
            "notools",
            "nonewdevices",
            "locksettings"
          ],

have i added the folowing code into the common.js

            if (r == 'locksettings') { newAccRights |= 1024; }
            if (r == 'nonewdevices') { newAccRights |= 4096; }

After the change, it works as intended, new accounts no longer recive the "new Devices" and "account settigs" rights.

![image](https://github.com/Ylianst/MeshCentral/assets/159419580/85ffc527-2aa2-47cd-9de4-39dc47b119fd)

However, if the Server recives an update, these changes will be overwritten. Perhaps we can incorporate or adapt this modification into the projekt.

**Reply by si458** [reply]
are you talking about only showing/hiding the 'add devices' button in the web ui or certain users?
if thats the case then its just a feature request as that should be simple enough i would think
(open one in github issues and follow template)

**Reply by Domkra** [reply, marked as answer]
My plan was to have the marked checkmarks set automatically when creating a new user, like in the picture

![image](https://github.com/Ylianst/MeshCentral/assets/159419580/464a33c3-be0f-4769-b303-dc5daacf9818)
